packages/driver/src/cy/net-stubbing/static-response-utils.ts

```diff
@@ -112,7 +112,16 @@ export function getBackendStaticResponse (staticResponse: Readonly<StaticRespons
       backendStaticResponse.body = staticResponse.body
     } else {
       backendStaticResponse.body = JSON.stringify(staticResponse.body)
-      _.set(backendStaticResponse, 'headers.content-type', 'application/json')
+
+      // There are various json-related MIME types. We cannot simply set it as `application/json`.
+      // @see https://www.iana.org/assignments/media-types/media-types.xhtml
+      if (
+        !(backendStaticResponse.headers &&
+        backendStaticResponse.headers['content-type'] &&
+        backendStaticResponse.headers['content-type'].includes('json'))
+      ) {
```

Comment:
Well, we might as well let people use any `content-type`. If they pass one we shouldn't ignore it in any case, it's extra confusing that way.

`caseInsensitiveGet` is required here, since the header names come from the user, they will not necessarily be lowercased.

Reply:
254ec0a

Reply:
@sainthkh there seem to be some failures with my change and `res.send()`, can you untangle it? it's something to do with the response merging

Reply:
You told me about the 2 issues:

1. We should let users set any `content-type`. 
2. We need `caseInsensitiveGet` because users' header names would not always be lowercased. 

I agree with 2. And fixed it with `caseInsensitiveGet`.

As for 1, I took the hybrid way: Throw an error when the user-provided `content-type` doesn't contain `json`. Set it to `application/json` when it's `undefined`.